src/modules/10-common/components/ModeSelector/__tests__/ModeSelector.test.tsx

```diff
@@ -0,0 +1,277 @@
+        CODE: {
+          icon: 'code',
+          label: 'common.purpose.code.title',
+          homePageUrl: '/account/px7xd_BFRCi-pfWPYXVjvw/module/code?noscope=true',
+          shouldVisible: true,
+          hasLicense: false,
+          color: '--code-border',
+          backgroundColor: '',
+          shortLabel: 'common.purpose.code.name'
+        },
+        IACM: {
+          icon: 'iacm',
+          label: 'common.iacmText',
+          homePageUrl: '/account/px7xd_BFRCi-pfWPYXVjvw/module/iacm?noscope=true',
+          shouldVisible: false,
+          hasLicense: false,
+          color: '--iacm-border',
+          backgroundColor: '',
+          shortLabel: 'common.infrastructures'
+        },
+        SSCA: {
+          icon: 'ssca-main',
+          label: 'common.sscaText',
+          homePageUrl: '/account/px7xd_BFRCi-pfWPYXVjvw/module/ssca?noscope=true',
+          shouldVisible: true,
+          hasLicense: false,
+          color: '--default-module-border',
+          backgroundColor: '',
+          shortLabel: 'common.sscaShortLabel',
+          moduleIntro: 'common.moduleIntro.softwareSupplyChainAssurance'
+        },
+        IDP: {
+          icon: 'idp',
+          label: 'common.purpose.idp.fullName',
+          homePageUrl: '/account/px7xd_BFRCi-pfWPYXVjvw/module/idp/default',
+          shouldVisible: true,
+          hasLicense: false,
+          color: '--idp-border',
+          backgroundColor: '',
+          shortLabel: 'common.purpose.idp.name',
+          moduleIntro: 'common.moduleIntro.idp'
+        },
+        CET: {
+          icon: 'cet',
+          label: 'common.purpose.cet.continuous',
+          homePageUrl: '/account/px7xd_BFRCi-pfWPYXVjvw/module/cet?noscope=true',
+          shouldVisible: true,
+          hasLicense: false,
+          color: '--cet-border',
+          backgroundColor: '',
+          shortLabel: 'common.purpose.errorTracking.title',
+          moduleIntro: 'common.moduleIntro.continuousErrorTracking'
+        },
+        SEI: {
+          icon: 'sei-main',
+          label: 'common.purpose.sei.fullName',
+          homePageUrl: '/account/px7xd_BFRCi-pfWPYXVjvw/module/sei?noscope=true',
+          shouldVisible: false,
+          hasLicense: false,
+          color: '--default-module-border',
+          backgroundColor: '',
+          shortLabel: 'common.purpose.sei.continuous',
+          moduleIntro: 'common.moduleIntro.insights'
+        }
+      }
+    },
+    DEFAULT_MODULES_ORDER: ['CD']
+  }
+})
+
+describe('Mode selector', () => {
+  beforeEach(async () => {
+    jest.clearAllMocks()
+    ;(usePreferenceStore as jest.Mock).mockImplementation(() => {
+      return {
+        setPreference: jest.fn(),
+        preference: {
+          orderedModules: ['CD', 'CI', 'CV', 'CF', 'CE', 'STO', 'CHAOS', 'CODE', 'SSCA', 'IDP', 'CET'],
+          selectedModules: ['CD', 'CI', 'CV', 'CF', 'CE', 'STO', 'CHAOS', 'CODE', 'SSCA', 'IDP', 'CET']
+        },
+        clearPreference: jest.fn
```

Comment:
```suggestion
        clearPreference: jest.fn()
```

since its a function

Reply:
Done